File: Lenguaje de Programación 2/Final/ejemploMonitores/ejemploMonitores/src/ejemplomonitores/recurso.java
package ejemplomonitores;
public class recurso {
    public int cantidad;
    public recurso(){
        System.out.println("Se ha creado el recurso");
        this.cantidad = 0;
        System.out.println("Cantidad: "+ this.cantidad);
    }
    public synchronized void producir() throws Exception{
        while (this.cantidad == 2){
            wait();
        }
        System.out.println("Se está produciendo...");
        int temp = this.cantidad;
        temp = temp + 1;
        this.cantidad = temp;
        System.out.println("Cantidad: "+ this.cantidad);
        notifyAll();
    }
    public synchronized void consumir() throws Exception{
        while (this.cantidad == 0){
            wait();
        }
        System.out.println("Se está consumiendo...");
        int temp = this.cantidad;
        temp = temp - 1;
        this.cantidad = temp;
        System.out.println("Cantidad: "+ this.cantidad);
        notifyAll();
    }
}

File: Lenguaje de Programación 2/Final/MonitoresC#/ConsProd/ConsProd/Recurso.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace ConsProd
{
    public class Recurso
    {
        private int cantidad;

        public Recurso()
        {
            System.Console.WriteLine("Se ha creado el recurso");
            this.cantidad = 0;
            System.Console.WriteLine("Cantidad: " + this.cantidad);
        }

        public void producir()
        {
            lock(this){
                while (this.cantidad == 2) Monitor.Wait(this);
                System.Console.WriteLine("Se está produciendo...");
                int temp = this.cantidad;
                temp = temp + 1;
                this.cantidad = temp;
                System.Console.WriteLine("Cantidad: " + this.cantidad);
                Monitor.PulseAll(this);
            }
        }

        public void consumir()
        {
            lock (this)
            {
                while (this.cantidad == 0) Monitor.Wait(this);
                System.Console.WriteLine("Se está consumiendo...");
                int temp = this.cantidad;
                temp = temp - 1;
                this.cantidad = temp;
                System.Console.WriteLine("Cantidad: " + this.cantidad);
                Monitor.PulseAll(this);
            }
        }
    }
}
